library/datasort.c

```diff
@@ -560,6 +561,80 @@ static int datasort_split_iterator_free(struct eblob_iterate_control *ictl __att
 	return 0;
 }
 
+static int datasort_load_index(struct datasort_cfg *dcfg,
+			       struct eblob_base_ctl *bctl,
+			       struct datasort_chunk *chunk) {
+	// Sanity
+	assert(dcfg);
+	assert(bctl);
+	assert(chunk);
+	assert(chunk->index == NULL); // Assume that index is already empty and we don't need clear previous data
+
+	int err;
+	struct eblob_file_ctl index_ctl = bctl->index_ctl;
+	assert(index_ctl.fd > 0);
+
+	chunk->index = calloc(index_ctl.size, sizeof(struct eblob_disk_control));
```

Comment:
Is it correct size that should be allocated?

Reply:
You right. size should be divided by sizeof of the structure.